File: solutions/swift/raindrops/1/Sources/Raindrops/Raindrops.swift
//Solution goes in Sources

class Raindrops {
    private var _number: Int
    
    init(_ number: Int) {
        _number = number
    }
    
    var sounds: String {
        get {
            var ret = ""
            if (_number % 3 == 0) {ret += "Pling"}
            if (_number % 5 == 0) {ret += "Plang"}
            if (_number % 7 == 0) {ret += "Plong"}
            if (ret.count == 0) {ret = "\(_number)"}
            return ret
        }
    }
    
}

File: solutions/prolog/raindrops/1/raindrops.pl
convert(N, Sounds) :-
    (0 is N mod 3 -> Pling = "Pling" ; Pling = ""),
    (0 is N mod 5 -> Plang = "Plang" ; Plang = ""),
    (0 is N mod 7 -> Plong = "Plong" ; Plong = ""),
    (Pling = Plang, Plang = Plong, Pling = ""
      -> number_string(N, Sounds)
      ; string_concat(Pling, Plang, Temp),
        string_concat(Temp, Plong, Sounds)
    ).
